common/codec/version0_thriftrw_test.go

```diff
@@ -84,6 +88,35 @@ func (s *thriftRWEncoderSuite) TestEncode() {
 	s.Equal(thriftEncodedBinary, binary)
 }
 
+func (s *thriftRWEncoderSuite) TestEncodeConcurrent() {
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	var finalErr error
+	count := 200
+	wg.Add(count)
+	for i := 0; i < count; i++ {
+		go func() {
+			defer wg.Done()
+			binary, err := s.encoder.Encode(thriftObject)
+			if err != nil {
+				mu.Lock()
+				finalErr = multierr.Append(finalErr, err)
+				mu.Unlock()
+				return
+			}
+
+			if diff := cmp.Diff(thriftEncodedBinary, binary); diff != "" {
+				mu.Lock()
+				finalErr = multierr.Append(finalErr, fmt.Errorf("Mismatch (-want +got):\n%s", diff))
+				mu.Unlock()
+				return
+			}
+		}()
+	}
+	wg.Wait()
+	s.NoError(finalErr)
```

Comment:
```suggestion
	var mu sync.Mutex
	count := 200
	errs := make([]error, count)
	wg.Add(count)
	for i := 0; i < count; i++ {
		i := i // until loop-var experiment stabilizes
		go func() {
			defer wg.Done()
			binary, err := s.encoder.Encode(thriftObject)
			if err != nil {
				errs[i] = err
				return
			}

			if diff := cmp.Diff(thriftEncodedBinary, binary); diff != "" {
				errs[i] = fmt.Errorf("Mismatch (-want +got):\n%s", diff))
				return
			}
		}()
	}
	wg.Wait()
	s.NoError(multierr.Append(errs...))
```
^ possibly a simpler pattern for future stuff

Reply:
correct one is `multierr.Combine(errs...)`

Reply:
yea, I'm not usually excited about nil-coalescing APIs.  there's no real naming pattern that identifies them :\

this one I've just seen a fair number of places because it's handy for reasons like this, where errors are just collected and not individually fatal.